paper.tex

```diff
@@ -287,17 +295,66 @@ \section{Masked Authenticate Messaging}
-
-The operations of MAM are frequent, especially for data providers, who either upload data in a short time interval or maintain multiple MAM channels or endpoints at the same time, hence the operation of MAM is one of the potential bottleneck in data marketplace. In this paper, we evaluate the performance of MAM, and improve the efficiency with different strategies. Also, we take offloading the computation tasks to brokers as another possible approach for IoT devices, since brokers are considered to have higher computational power.
+\subsubsection{A Traceable Data Stream}
+Data traceability is an important security issue that allows users to track the changes of data. Currently, hash is widely used to generate checksums of entire package of files, users can compare the hash value of files in order to check the integrity. Yet the hash values of each new version are unrelated and it's hard to point out the differences between versions. MAM benefits from the singly linked-list data structure which attaches messages chronologically, users can easily track the footprints of data change logs as well as checking the validity of modifications with the signature.
 
-\subsection{TA - endpoint}
+\subsection{Delegate MAM operations to Tangle-accelerator}
 \label{section:ta_endpoint}
+MAM builds a secure and authenticated communication protocol on top of multiple cryptosystems, which low-level devices need to spend a lot of resources to apply it, and they may not even support built-in libraries in regular operating system. To overcome these concerns, we propose another approach that allows data providers to delegate the MAM operations to Tangle-accelerators, proxy servers with high computation power, which can accelerates the interactions with Tangle while ensuring the privacy of data providers. Fig.~\ref{fig:delegation} shows how data providers can adopt MAM.
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=\linewidth]{delegation}
+    \caption{Data providers can use MAM in 2 ways: the provider performs MAM operations locally, and the provider delegates MAM operations to Tangle-accelerator.}
+    \label{fig:delegation}
+\end{figure}
+
+
+\subsubsection{Infrastructure}
+To build the system, MQTT\cite{MQTT} is chosen as the communication protocol. MQTT protocol utilizes much more lightweight packet than HTTP protocol has. The topic-based publish-subscribe pattern and lightweight packet that MQTT adopts would undoubtedly play an important role on low-level edge devices with limited bandwidth. The publish-subscribe model decouples publishers and subscribers in many ways:
```

Comment:
Replace the statement which mentioned "much more lightweight package" with "MQTT requires fewer TCP packets and less traffic."